io/io/src/TFile.cxx

```diff
@@ -1343,19 +1344,17 @@ std::pair<TList *, Int_t> TFile::GetStreamerInfoListImpl(bool lookupSICache)
 
 #ifdef R__USE_IMT
       if (lookupSICache) {
-         if (!fgTsSIHashes.Insert(buf,fNbytesInfo)) {
+         if (!fgTsSIHashes.Insert(bufferPtr,fNbytesInfo)) {
             if (gDebug > 0) Info("GetStreamerInfo", "The streamer info record for file %s has already been treated, skipping it.", GetName());
             return {nullptr, 0};
          }
       }
 #else
       (void) lookupSICache;
 #endif
-      key->ReadKeyBuffer(buf);
-      list = dynamic_cast<TList*>(key->ReadObjWithBuffer(buffer));
+      key->ReadKeyBuffer(bufferPtr);
+      list = dynamic_cast<TList*>(key->ReadObjWithBuffer(buffer.get()));
```

Comment:
This took me a while... Here, you need to put the *original* pointer to the buffer because the `key::ReadKeyBuffer` calls `frombuf`, which modifies the pointer itself (`bufferPtr` in this case). Same reason why the original code hat the `buf` and `buffer` variables.

Reply:
However, valgrind after the changes (same than before):

```
==6390== LEAK SUMMARY:
==6390==    definitely lost: 6,424 bytes in 202 blocks
==6390==    indirectly lost: 23,936 bytes in 200 blocks
==6390==      possibly lost: 60,990 bytes in 608 blocks
==6390==    still reachable: 74,910,348 bytes in 100,853 blocks
==6390==                       of which reachable via heuristic:
==6390==                         newarray           : 25,424 bytes in 49 blocks
==6390==                         multipleinheritance: 2,136 bytes in 3 blocks
==6390==         suppressed: 6,375,935 bytes in 65,625 blocks
```

Reply:
I think we should add a comment to help the future reader of the code.